src/main/scala/com/fulcrumgenomics/fasta/CollectAlternateContigNames.scala

```diff
@@ -213,7 +216,18 @@ class CollectAlternateContigNames
 
     // Apply to an existing sequence dictionary if necessary
     val dict: SequenceDictionary = existing match {
-      case None => SequenceDictionary(metadatas.toSeq:_*)
+      case None       =>
+        if (!sortBySequencingRole) SequenceDictionary(metadatas.toSeq:_*) else {
+          // Use the user-provided roles, otherwise use all role
+          val roles = (if (sequenceRoles.nonEmpty) sequenceRoles.toIndexedSeq else SequenceRole.values)
+            .zipWithIndex
+            .map { case (role, index) =>  role.key -> index }
+            .toMap
+          val metas = metadatas.toSeq.sortBy { metadata =>
+            metadata.get(Column.SequenceRole.tag).map { name => roles(name) }.getOrElse(roles.size)
+          }.zipWithIndex.map { case (metadata, index) => metadata.copy(index=index) }
+          SequenceDictionary(metas:_*)
+        }
```

Comment:
Unit tests recommended for the new code branches you created including whether `sequenceRoles` is defined or not. You have `roles(name)` instead of `roles.get(name)` so I'm assuming code earlier guarantees the metadata doesn't have any invalid role names?